kernel/src/platform/mpu.rs

```diff
@@ -1,105 +1,193 @@
+    /// `initial_app_memory_size`   : initial size of app-owned memory
+    /// `initial_kernel_memory_size`: initial size of kernel-owned memory
+    /// `permissions`               : permissions for the MPU region
+    /// `config`                    : MPU region configuration
+    ///
+    /// # Return Value
+    ///
+    /// This function returns the start address and the size of the memory block
+    /// chosen for the process. If it is infeasible to find a memory block or
+    /// allocate the MPU region, or if the function has already been called, returns
+    /// None.
+    #[allow(unused_variables)]
+    fn allocate_app_memory_region(
+        &self,
+        unallocated_memory_start: *const u8,
+        unallocated_memory_size: usize,
+        min_memory_size: usize,
+        initial_app_memory_size: usize,
+        initial_kernel_memory_size: usize,
+        permissions: Permissions,
+        config: &mut Self::MpuConfig,
+    ) -> Option<(*const u8, usize)> {
+        let memory_size = {
+            if min_memory_size < initial_app_memory_size + initial_kernel_memory_size {
+                initial_app_memory_size + initial_kernel_memory_size
+            } else {
+                min_memory_size
+            }
+        };
+        if memory_size > unallocated_memory_size {
+            None
+        } else {
+            Some((unallocated_memory_start, memory_size))
+        }
+    }
+
+    /// Updates the MPU region for app-owned memory.
+    ///
+    /// An implementation must reallocate the MPU region for app-owned memory stored in
+    /// `config` to maintain the 3 conditions described in `allocate_app_memory_region`.
+    ///
+    /// # Arguments
+    ///
+    /// `app_memory_break`      : new address for the end of app-owned memory
+    /// `kernel_memory_break`   : new address for the start of kernel-owned memory
+    /// `permissions`           : permissions for the MPU region
+    /// `config`                : MPU region configuration
+    ///
+    /// # Return Value
+    ///
+    /// Returns an error if it is infeasible to update the MPU region, or if it was
+    /// never created.
+    #[allow(unused_variables)]
+    fn update_app_memory_region(
+        &self,
+        app_memory_break: *const u8,
+        kernel_memory_break: *const u8,
+        permissions: Permissions,
+        config: &mut Self::MpuConfig,
+    ) -> Result<(), ()> {
+        if (app_memory_break as usize) > (kernel_memory_break as usize) {
+            Err(())
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Configures the MPU with the provided region configuration.
+    ///
+    /// An implementation must ensure that all memory locations not covered by
+    /// an allocated region are inaccessible in user mode and accessible in
+    /// supervisor mode.
+    ///
+    /// # Arguments
+    ///
+    /// `config`    : MPU region configuration
+    #[allow(unused_variables)]
```

Comment:
use underscore instead i.e ```fn configure_mpu(&self, _config: &Self::MpuConfig)```

Reply:
I was actually imitating `kernel/src/driver.rs` on this -- it also uses this attribute for its default implementations.

Reply:
ok, you don't have to address it just a minor thing.....